packages/client/_src/Resources/Blog/CreatePost.ts

```diff
@@ -0,0 +1,9 @@
+import { BlogPost, BlogPostId } from "@effect-ts-app/boilerplate-types/Blog"
+
+@allowAnonymous
+@allowRoles("user")
+export class CreatePostRequest extends Post("/blog/posts")<CreatePostRequest>()(
+  BlogPost.pick("title", "body")
+) {}
+
+export const CreatePostResponse = BlogPostId
```

Comment:
What is the role of this export?

Does every `Resource` has to export a `FooRequest` and `FooResponse`?

Why?

Reply:
The role is to provide a Schema for serialising the response to JSON, so that no wrong types or excess properties go out. It makes the Controllers return typed, and the client action returns typed.

No, if you don't export a Response, the response is defaulting to `Void`, and nothing will be serialised back to the response.